src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/service/impl/MessageProcessorServiceImpl.java

```diff
@@ -37,33 +38,42 @@
     @Autowired
     private MessageProducer messageProducer;
 
+    @Autowired
+    private ConsumerConfig consumerConfig;
+
     @Override
-    public void processMessage(ChipsRestInterfacesSend message) throws ServiceException {
+    public void processMessage(ChipsRestInterfacesSend message, boolean isErrorAttempt) throws ServiceException {
         Map<String, Object> logMap = new HashMap<>();
         logMap.put("Message", message.getData());
         try {
             chipsRestClient.sendToChips(message);
         } catch (HttpStatusCodeException hsce) {
             logMap.put("HTTP Status Code", hsce.getStatusCode().toString());
-            handleFailedMessage(message, hsce, logMap);
+            handleFailedMessage(message, hsce, logMap, isErrorAttempt);
         } catch (Exception e) {
-            handleFailedMessage(message, e, logMap);
+            handleFailedMessage(message, e, logMap, isErrorAttempt);
         }
     }
 
-    private void handleFailedMessage(ChipsRestInterfacesSend message, Exception e, Map<String, Object> logMap) throws ServiceException {
+    private void handleFailedMessage(ChipsRestInterfacesSend message, Exception e, Map<String, Object> logMap, boolean isErrorAttempt) throws ServiceException {
         var messageId = message.getMessageId();
         logger.error(String.format(SEND_FAILURE_MESSAGE, messageId), e, logMap);
 
+        if (isErrorAttempt) {
+            message.setAttempt(1);
+            messageProducer.writeToTopic(message, consumerConfig.retryTopic());
+            return;
+        }
+
         var attempts = message.getAttempt();
         logger.info(String.format("Attempt %s failed for message id %s", attempts, messageId), logMap);
 
         if (attempts < maxRetryAttempts) {
             message.setAttempt(attempts + 1);
-            messageProducer.writeToTopic(message, retryTopicName);
+            messageProducer.writeToTopic(message, consumerConfig.retryTopic());
```

Comment:
was looking at the retry method here - https://github.com/companieshouse/ch-kafka/blob/6dabe9394a901688b1a0b81501c839ec1c835f08/src/main/java/uk/gov/companieshouse/kafka/consumer/resilience/CHKafkaResilientConsumerGroup.java#L85
That one seems to set the timestamp anew, should we be setting a new timestamp on retries? seems like that is what ch-kafka does. and also maybe when we try and process one from error topic?

Reply:
or leave for another pr

Reply:
i'd rather leave this for a different pr, this is just about getting the error consumer to work correctly.
Refactoring our retryto use the inbuilt retry method should probably be its own pr